src/fsharp/NameResolution.fs

```diff
@@ -763,26 +763,81 @@ let AddStaticContentOfTyconRefToNameEnv (g:TcGlobals) (amap: Import.ImportMap) m
-                   |> List.groupBy (fun m -> m.PropertyName)
-               
-               for (propName, propInfos) in propInfos do
-                   let propInfos = propInfos |> List.filter (fun m -> m.IsStatic)
-                   for propInfo in propInfos do 
-                       yield KeyValuePair(propName , Item.Property(propName,[propInfo])) |]
-
-        { nenv with eUnqualifiedItems = nenv.eUnqualifiedItems.AddAndMarkAsCollapsible items }
+            // This is to handle the current behavior with qualified extension members:
+            //     Extension member properties will hide all extension methods that share the same name.
+            [|
+                let propItems = getPropItems ()
+                let methItems = getMethItems ()
+
+                let itemGroups =
+                    propItems @ methItems
+                    |> List.groupBy (fun pair -> pair.Key)
+
+                for (_, items) in itemGroups do
+                    let sortedItems =
+                        items 
+                        |> List.sortBy (fun pair -> 
+                            match pair.Value with 
+                            | Item.Property (_, [propInfo]) -> not propInfo.IsExtensionMember 
+                            | Item.MethodGroup (_, methGroup, _) -> methGroup |> List.exists (fun x -> not x.IsExtensionMember)
+                            | _ -> false)
+                    match sortedItems with
+                    | item :: _ -> yield item
+                    | _ -> ()
+            |]
+
+        { nenv with 
+            eUnqualifiedItems = 
+                (nenv.eUnqualifiedItems, items) 
+                ||> Array.fold (fun x (KeyValue(k, v)) -> 
+                    match x.TryGetValue k with
+                    | true, existingItem ->
+                        match existingItem, v with
+                        
+                        // Combine methods groups.
+                        | Item.MethodGroup (_, minfosCurrent, None), Item.MethodGroup (_, minfosNew, None) ->
+                            let minfos =
+                                (minfosNew, minfosCurrent)
+                                ||> List.fold (fun minfos minfo1 ->
+                                    // Shadow methods with the same signature.
+                                    if minfos |> List.exists (fun minfo2 -> MethInfosEquivByNameAndSig Erasure.EraseAll true g amap m minfo1 minfo2) then
+                                        minfos
+                                    else
+                                        minfo1 :: minfos
+                                )
+                            x.Add(k, Item.MethodGroup (k, minfos, None))
+
+                        // Methods will hide properties, events, and fields with the same name.
+                        | Item.MethodGroup _, Item.Property _
```

Comment:
This seems fine, will need test cases and RFC update

Reply:
Yup, for sure. This is to mimic opening multiple static classes in C# and what their behavior is.